python-libraries/data-platform-catalogue/data_platform_catalogue/entities_v2.py

```diff
@@ -0,0 +1,272 @@
+        description="Location of the data in s3", default=None
+    )
+
+
+# Needs refining
+class DataSummary(BaseModel):
+    """
+    Summarised information derived from the actual data.
+    """
+
+    row_count: int | None = Field(
+        description="Row count when the metadata was last updated", default=None
+    )
+
+
+class Asset(BaseModel):
+    """
+    Any searchable data asset that is present in the metadata graph, which
+    may be related to other assets.
+    Examples include platforms, databases, tables, data products
+    """
+
+    id: str | None = Field("Unique identifier for the asset. Relates to Datahub's urn")
+    display_name: str | None = Field("Display name of the asset")
+    name: str = Field("Actual name of the asset in its source platform")
+    fully_qualified_name: str | None = Field(
+        "Fully qualified name of the asset in its source platform"
+    )
+    description: str = Field(
+        description="Detailed description about what functional area this asset is representing, what purpose it has and business related information.",
+    )
+    relationships: dict[RelationshipType, list[AssetRef]] | None = Field(
+        default=None,
+        description="References to related assets in the metadata graph, such as platform or parent assets",
+    )
+    domain: DomainRef = Field(description="The domain this asset belongs to.")
+    governance: Governance = Field(description="Information about governance")
+    usage_restrictions: UsageRestrictions = Field(
+        description="Limitations on how the data may be used and accessed"
+    )
+    tags: list[TagRef] = Field(
+        default_factory=list,
+        description="Additional tags to add.",
+    )
+
+    # Needs refining
+    last_updated: datetime = Field(
+        description="When the metadata was last updated in the catalogue"
+    )
+    first_created: datetime = Field(description="When the data asset was first created")
+
+
+class Database(Asset):
+    """
+    For source system databases
+    """
+
+    status: DatabaseStatus = Field(
+        default=DatabaseStatus.PROD,
+        description="Whether this database represents production data or not",
+    )
+
+
+class Table(Asset):
+    """
+    A table in a database or a tabular dataset
+    """
+
+    column_details: list[Column] = Field(
+        description="A list of objects which relate to columns in your data, each list item will contain, a name of the column, data type of the column and description of the column."
+    )
+    fully_qualified_name: str | None = Field(
+        default=None,
+        description="Fully qualified table name as it appears in the source platform, including the database",
+    )
+    access_information: AccessInformation = Field(
+        description="Metadata about how to access the data"
+    )
+    data_summary: DataSummary = Field(
+        description="Summary of data stored in this table"
+    )
+
+
+class Chart(Asset):
```

Comment:
Suggest changing asset to entity, as it aligns more closely with the meaning, and [DataHub's](https://datahubproject.io/docs/metadata-modeling/metadata-model) [terminology](https://datahubproject.io/docs/metadata-modeling/metadata-model/#the-core-entities)
> For example, an instance of a Dataset or a CorpUser is an Entity
```suggestion
class Entity(BaseModel):
    """
    Any searchable data entity that is present in the metadata graph, which
    may be related to other entities.
    Examples include platforms, databases, tables, data products
    """

    id: str | None = Field("Unique identifier for the entity. Relates to Datahub's urn")
    display_name: str | None = Field("Display name of the entity")
    name: str = Field("Actual name of the entity in its source platform")
    fully_qualified_name: str | None = Field(
        "Fully qualified name of the entity in its source platform"
    )
    description: str = Field(
        description="Detailed description about what functional area this entity is representing, what purpose it has and business related information.",
    )
    relationships: dict[RelationshipType, list[EntityRef]] | None = Field(
        default=None,
        description="References to related entities in the metadata graph, such as platform or parent entities",
    )
    domain: DomainRef = Field(description="The domain this entity belongs to.")
    governance: Governance = Field(description="Information about governance")
    usage_restrictions: UsageRestrictions = Field(
        description="Limitations on how the data may be used and accessed"
    )
    tags: list[TagRef] = Field(
        default_factory=list,
        description="Additional tags to add.",
    )

    # Needs refining
    last_updated: datetime = Field(
        description="When the metadata was last updated in the catalogue"
    )
    first_created: datetime = Field(description="When the data entity was first created")


class Database(Entity):
    """
    For source system databases
    """

    status: DatabaseStatus = Field(
        default=DatabaseStatus.PROD,
        description="Whether this database represents production data or not",
    )


class Table(Entity):
    """
    A table in a database or a tabular dataset
    """

    column_details: list[Column] = Field(
        description="A list of objects which relate to columns in your data, each list item will contain, a name of the column, data type of the column and description of the column."
    )
    fully_qualified_name: str | None = Field(
        default=None,
        description="Fully qualified table name as it appears in the source platform, including the database",
    )
    access_information: AccessInformation = Field(
        description="Metadata about how to access the data"
    )
    data_summary: DataSummary = Field(
        description="Summary of data stored in this table"
    )


class Chart(Entity):
```